daemon/daemon_unix.go

```diff
@@ -1184,3 +1194,30 @@ func setupOOMScoreAdj(score int) error {
 	f.Close()
 	return err
 }
+
+func (daemon *Daemon) initCgroupsPath(path string) error {
+	if path == "/" || path == "." {
+		return nil
+	}
+
+	daemon.initCgroupsPath(filepath.Dir(path))
+
+	root := "/sys/fs/cgroup/cpu"
+	path = filepath.Join(root, path)
+	sysinfo := sysinfo.New(false)
+	if err := os.MkdirAll(path, 0755); err != nil && !os.IsExist(err) {
+		return err
+	}
+	if sysinfo.CPURealTimePeriod && daemon.configStore.CPURealTimePeriod != 0 {
```

Comment:
`runC` does something similar for cgroups for the container, it does not ensure parent groups have the proper value set.

This logic basically solves the issue of `runC` failing to set the `cpu.rt_runtime` of the container because its parent value is 0.